File: 0463.岛屿的周长/0463-岛屿的周长.rs
impl Solution {
    pub fn island_perimeter(grid: Vec<Vec<i32>>) -> i32 {
        let r = grid.len();
        let c = grid[0].len();
        let mut cnt = 0; 
        let mut num = 0; // the numbers of 1
        for i in 0..r {
            for j in 0..c {
                if grid[i][j] == 1 {
                    num += 1;
                }
            }
        }
        for i in 0..r {
            for j in 0..c-1 {
                if grid[i][j] == 1 && grid[i][j+1] == 1 {
                    cnt += 1;
                }
            }
        }
        for i in 0..c {
            for j in 0..r-1 {
                if grid[j][i] == 1 && grid[j+1][i] == 1 {
                    cnt += 1;
                }
            }
        }
        
        num*4 - cnt*2
    }
}

File: 0463.岛屿的周长/0463-岛屿的周长.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        cnt = 0  # 挨着的边
        num = 0  # numbers of 1
        r, c = len(grid), len(grid[0])
        for i in range(r):
            for j in range(c):
                if grid[i][j] == 1:
                    num += 1
        for i in range(r):
            for j in range(c - 1):
                if grid[i][j] == 1 and grid[i][j + 1] == 1:
                    cnt += 1
        for i in range(c):
            for j in range(r - 1):
                if grid[j][i] == 1 and grid[j + 1][i] == 1:
                    cnt += 1

        return 4 * num - 2 * cnt
